main.cpp

```diff
@@ -0,0 +1,147 @@
+#include <iostream>
+#include <vector>
+#include <algorithm>
+
+class sudoku
+{
+protected:
+    int sudokmatrix[9][9];
+    std::vector<int> Possibilities;
+    bool findEmptySlot(int& row, int& col);
+    bool canPlaceNum(int row, int col, int num);
+    bool numAlreadyInRow(int row, int num);
+    bool numAlreadyInCol(int col, int num);
+    bool numAlreadyInBox(int smallGridRow, int smallGridCol, int num);
+public:
+    sudoku();
+    ~sudoku() {};
+    void enterDigits();
+    void print();
+    bool solve();
+};
+
+
+sudoku::sudoku()
```

Comment:
For zero initialization of sudokmatrix you can use
sudoku::sudoku() : sudokmatrix{} {
   std::cout << "\n matrix initialized \n";
}